### PR TITLE
feat(core): add account center api

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Add GET/PATCH api of account center, this settings will be used the control the global access of Profile API.

## Testing

Integration tests.

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [x] integration tests
- [x] necessary TSDoc comments

### CODE REVIEW

File: packages/core/src/queries/account-center.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+};
+
+export class AccountCenterQueries {
+  public readonly insert = buildInsertIntoWithPool(this.pool)(AccountCenters, {
+    returning: true,
+  });
+
+  public readonly update = buildUpdateWhereWithPool(this.pool)(AccountCenters, true);
+
+  public readonly find = buildFindEntityByIdWithPool(this.pool)(AccountCenters);
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/core/src/queries/account-center.ts]: updated

File: packages/core/src/queries/account-center.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { buildFindEntityByIdWithPool } from '../database/find-entity-by-id.js';
+import { buildUpdateWhereWithPool } from '../database/update-where.js';
+
+const { table, fields } = convertToIdentifiers(AccountCenters);
+
+const defaultAccountCenter: Omit<AccountCenter, 'tenantId'> = {
+  id: 'default',
+  enabled: false,
+  fields: {},
+};
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/core/src/queries/account-center.ts]: updated

File: packages/core/src/queries/account-center.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    const accountCenter = await this.pool.maybeOne<AccountCenter>(sql`
+      select * from ${table}
+      where ${fields.id} = ${defaultAccountCenter.id}
+    `);
+
+    if (accountCenter) {
+      return accountCenter;
+    }
+
+    await this.insert(defaultAccountCenter);
```
[the remark this answers is not in the document]
Comment (wangsijie) [reply packages/core/src/queries/account-center.ts]: updated

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**